src/main/java/org/opensearch/neuralsearch/processor/normalization/ZScoreNormalizationTechnique.java

```diff
@@ -0,0 +1,258 @@
+        switch (combinationTechnique.techniqueName()) {
+            case ArithmeticMeanScoreCombinationTechnique.TECHNIQUE_NAME:
+                // This is the supported technique, so we do nothing
+                break;
+            case GeometricMeanScoreCombinationTechnique.TECHNIQUE_NAME:
+                throw new IllegalArgumentException("Z Score does not support geometric mean combination technique");
+            case HarmonicMeanScoreCombinationTechnique.TECHNIQUE_NAME:
+                throw new IllegalArgumentException("Z Score does not support harmonic mean combination technique");
+            default:
+                throw new IllegalArgumentException(
+                    "Z Score does not support the provided combination technique {}: Supported technique is arithmetic_mean"
+                        + combinationTechnique.techniqueName()
+                );
+        }
+    }
+
+    @Override
+    public String describe() {
+        return String.format(Locale.ROOT, "%s", TECHNIQUE_NAME);
+    }
+
+    @Override
+    public Map<DocIdAtSearchShard, ExplanationDetails> explain(final List<CompoundTopDocs> queryTopDocs) {
+        ZScoreNormalizationTechnique.ZScores zScores = getZScoreResults(queryTopDocs);
+
+        Map<DocIdAtSearchShard, List<Float>> normalizedScores = new HashMap<>();
+        for (CompoundTopDocs compoundQueryTopDocs : queryTopDocs) {
+            if (Objects.isNull(compoundQueryTopDocs)) {
+                continue;
+            }
+            List<TopDocs> topDocsPerSubQuery = compoundQueryTopDocs.getTopDocs();
+            int numberOfSubQueries = topDocsPerSubQuery.size();
+            for (int subQueryIndex = 0; subQueryIndex < numberOfSubQueries; subQueryIndex++) {
+                TopDocs subQueryTopDoc = topDocsPerSubQuery.get(subQueryIndex);
+                for (ScoreDoc scoreDoc : subQueryTopDoc.scoreDocs) {
+                    DocIdAtSearchShard docIdAtSearchShard = new DocIdAtSearchShard(scoreDoc.doc, compoundQueryTopDocs.getSearchShard());
+                    float normalizedScore = normalizeSingleScore(
+                        scoreDoc.score,
+                        zScores.stdPerSubquery[subQueryIndex],
+                        zScores.meanPerSubQuery[subQueryIndex],
+                        zScores.maxPerSubQuery[subQueryIndex],
+                        zScores.minPerSubQuery[subQueryIndex]
+                    );
+                    ScoreNormalizationUtil.setNormalizedScore(
+                        normalizedScores,
+                        docIdAtSearchShard,
+                        subQueryIndex,
+                        numberOfSubQueries,
+                        normalizedScore
+                    );
+                    scoreDoc.score = normalizedScore;
+                }
+            }
+        }
+        return getDocIdAtQueryForNormalization(normalizedScores, this);
+    }
+
+    private int getNumOfSubqueries(final List<CompoundTopDocs> queryTopDocs) {
```

Comment:
Can we move this method to a util class class like HybridQueryUtil or ProcessorUtil? This logic is used by almost every normalization technique. We can reuse the same in every class.